packages/drawer/src/lib/foundation.tsx

```diff
@@ -28,8 +27,10 @@ const useDrawerFoundationFactory = (
       foundation: ({ rootEl, emit, getProps }) => {
         let previousFocusEl: HTMLElement;
 
-        const f = new MDCConstructor({
-          addClass: (className: string) => rootEl.addClass(className),
+        return new MDCConstructor({
+          //This delayes adding the class until next animation frame, fixes issue with draw getting stuck in open state
+          addClass: (className: string) =>
+            window.requestAnimationFrame(() => rootEl.addClass(className)),
```

Comment:
Does this work:
`addClass: (className: string) => rootEl.ref?.classList.add(className),`

Reply:
Yes it seems so, but I of course would need update all methods, e.g.:

`
         addClass: (className: string) => rootEl.ref?.classList.add(className),
         removeClass: (className: string) => rootEl.ref?.classList.remove(className),
         hasClass: (className: string) => rootEl.ref?.classList.contains(className) || false,
`